src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Symbols/SymbolManagerBase.cs

```diff
@@ -337,16 +331,18 @@ public bool Equals(TypeArrayKey other)
                 if (other._types == _types) return true;
                 if (other._types.Length != _types.Length) return false;
                 if (other._hashCode != _hashCode) return false;
-                for (int i = 0, n = _types.Length; i < n; i++)
+                for (int i = 0; i < _types.Length; i++)
                 {
                     if (!_types[i].Equals(other._types[i]))
                         return false;
                 }
                 return true;
             }
 
+            [ExcludeFromCodeCoverage] // Typed overload should always be the method called.
```

Comment:
That last point is precisely why I like that attribute, in that IMO it should only be on things that both should not be removed *and* should not be reachable, upping the odds that anything that's red in the coverage at the very least reflects a testing gap, and is more than likely to hide some bug. It's related to why I always add a comment when I use them; it marks that I have indeed addressed their not being used and my conclusion in doing so was that they should be left alone.